packages/launchpad-ledger/src/ledgerwallet.ts

```diff
@@ -0,0 +1,45 @@
+import { AccountData, encodeSecp256k1Signature, OfflineSigner, StdSignature } from "@cosmjs/launchpad";
+
+import { LaunchpadLedger } from "./launchpadledger";
+
+interface LedgerWalletOptions {
+  readonly testModeAllowed: boolean;
+}
+
+export class LedgerWallet implements OfflineSigner {
+  private readonly ledger: LaunchpadLedger;
+  private address: string | undefined;
+  private pubkey: Uint8Array | undefined;
+
+  constructor(options?: LedgerWalletOptions) {
+    this.ledger = new LaunchpadLedger(options);
+  }
+
+  public async getAccounts(): Promise<readonly AccountData[]> {
+    await this.ledger.connect();
+
+    const address = (this.address = this.address || (await this.ledger.getCosmosAddress()));
+    const pubkey = (this.pubkey = this.pubkey || (await this.ledger.getPubKey()));
```

Comment:
So here we could potentially return multiple accounts but there is no state available telling us which account exists, right?

Reply:
What do you mean by "exists"? Eg exists where?

Reply:
"exists" as in _has be created by the user before_.

A Ledger app can give us any address `m/44'/118'/0'/0/a` and from an `OfflineSigner.getAccounts` implementation I would expect to get all used accounts. However, I assume this is not possible without online [account discovery](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#Account_discovery). So we probably have to live with a single-account implementation.

What we could do is to accept a list of paths in the constructor instead of a single one.